merlin/systems/triton/conversions.py

```diff
@@ -42,31 +42,98 @@
 
 
 def triton_request_to_dict_array(request, column_names):
-    input_tensors = {
-        name: pb_utils.get_input_tensor_by_name(request, name).as_numpy() for name in column_names
-    }
-    return DictArray(input_tensors)
+    """
+    Turns a Triton request into a DictArray by extracting individual tensors
+    from the request using pb_utils.
+
+    Parameters
+    ----------
+    request : TritonInferenceRequest
+        The incoming request for predictions
+    column_names : List[str]
+        List of the input columns to extract from the request
+
+    Returns
+    -------
+    DictArray
+        Dictionary-like representation of the input columns
+    """
+    dict_inputs = {}
+    for name in column_names:
+        try:
+            values = _get_triton_tensor(request, f"{name}__values")
+            lengths = _get_triton_tensor(request, f"{name}__lengths")
```

Comment:
Here we establish a convention for how we expect multi-array features to be formatted in the Triton request. Since Triton can only accept a flat list of named tensors, there's no way for us to group these together without some kind of naming convention.